### DISCUSSION TITLE
Adding Examine-Description Str. Attribute

### DISCUSSION THREAD

**Discussion by tajmone** [alan-if/alan-i18n#22] [Libraries Changes]
@thoni56, there's one feature of the StdLib which I _really_ love and think it deserves to be part of the Starter Libs: the `ex` string attribute for 'examine' descriptions.

Every `thing` is given an empty `ex` attribute (`Has ex "".`) which is then used by the 'eamine' verbs: if it's an empty string, the default "You notice nothing unusual..." message is printed, otherwise the `ex` string is printed.

This makes it very easy and quick to provide static description for 'examine', without having to implement a VERB EXAMINE on each instance — but of course, you can still do if you need to handle it dynamic, via conditional code, ecc.

```alan
The chair is object at lounge.
  Has xdesc "Just an old chair that's been here for ages.".
End the.
```

Examining things is such a core feature of any adventure that I believe this feature deserves to be included in the libraries (maybe I'd call the attribute `xdesc` rather than `ex`, for the former is a better mnemonic for "examine description"). Sure, it adds a bit of overhead since every single instance will have that attribute added, but it reduces author code, and I guess that empty string attributes are just _nil_ values (i.e. absence of some sort of "pointer"/offset to the actual string in the storyfile).

But of course, this would have to be a feature found on all libraries locales, so if you're OK with it I'll add it to the English and Italian libs, and you'll add it to the Swedish — future porters will most likely simply translate it from the English library. As for the Spanish library, I could try, but only if it doesn't involve having to change Spanish output text (don't want to mess up anything).

Until all libraries reach v1.0.0, we're probably going to need to coordinate efforts for new features, since we'll have to add them to our translation too. We'll probably need to coordinate such changes in some way, e.g. via a v1.0.0 Roadmap document, especially for the Spanish library which might end up waiting for a native speaker to enforce some updates required for v1.0.0.

**Comment by thoni56** [comment]
As you probably are aware, this was not the initial intention with string attributes. But it seems that "users" have found this to be a natural, working and handy mechanism to make it easy to manage authoring, allowing for larger and more sophisticated games.

So, I think it could definitely be one item on the 1.0.0 roadmap.

**Reply by tajmone** [reply]
# EN Lib v0.1.2: Implemented

OK @thoni56, I've now added the `xDes` attribute to StartLib EN `v0.1.2` (only bumped up PATCH, since it's a non-breaking change) and tweaked 'examine' accordingly.

I've also tweaked the _Cloak_ adventure to make use of it, where it was possible (most objects have dynamic 'examine' code, so they were not eligible for its use). Thanks to the new stress-test, I was able to verify that this new feature didn't affect the generated transcripts, which confirms that it was integrated smoothly.

I've also edited the `test.sh` script of Cloak to be UTF-8 ready, following your advice from alan-if/alan#32 on not using redirection for transcripts, but rename them after generation instead.
